src/core/style/_elevation.scss

```diff
@@ -0,0 +1,186 @@
+/**
+ * A collection of mixins and CSS classes that can be used to apply elevation to a material
+ * element.
+ * See: https://www.google.com/design/spec/what-is-material/elevation-shadows.html
+ * Examples:
+ *
+ * ```scss
+ * .md-foo {
+ *   @include $md-elevation(2);
+ *
+ *   &:active {
+ *     @include $md-elevation(8);
+ *   }
+ * }
+ * ```
+ *
+ * ```html
+ * <div id="external-card" class="md-elevation-z2"><p>Some content</p></div>
```

Comment:
Yes, I think you understand my question correctly. Let's take the standalone / bootstrap stylesheet for a moment out of the way.

I don't think Angular 2 scans the whole stylesheet for duplicate identifiers, that won't be even possible, since some selectors are just overriding a previous defined selector.

The point is, for example: we have the button component, which has view encapsulation disabled. ([Source](https://github.com/angular/material2/blob/master/src/components/button/button.ts#L30))

Then the buttons stylesheet will import the `_elevation.scss` file, this causes the `md-elevation-z*` classes to be generated.

So now imagine, if you have more than one component, which includes the elevation and has `ViewEncapsulation` disabled. This will cause several generations of the same classes.

In my opinion we should separate between a bootstrap / standalone stylesheet and the styles which are dynamically inserted into the DOM.

Removing those class generation would be also possible, since the mixins will only generate the specific elevation (thats awesome!)